### PR TITLE
Prevent infinite looping when following forward-links

### LINKED ISSUE #1894: Prevent infinite loop when following forward links
Take the `GetProof` method of the skipblock DB. It assumes forward links have been correctly added in the blocks but a recent bug was storing new blocks with previous forward links if they were present, which was creating infinite loop in the path as we don't verify the from and to when reading them.

Those infinite loops should be prevented.

### PR DESCRIPTION
**Pull request by Gilthoniel**
This adds a check in functions that are following forward-links
to insure that it moves to a more recent block every time and
return an error otherwise.

Also: check forward-links when storing a unknown block.
    
Fixes #1894

### CODE REVIEW

File: skipchain/struct.go [hunk 1]
```diff
@@ -767,6 +771,35 @@ func (db *SkipBlockDB) StoreBlocks(blocks []*SkipBlock) ([]SkipBlockID, error) {
 						return fmt.Errorf("Tried to store unlinkable block: %+v", sb.SkipBlockFix)
 					}
 				}
+
+				// As we don't store the target height in the forward-link, there is no
+				// way to insure that it will be stored at the right height if the target
```
Comment (kc1212) [inline skipchain/struct.go]: insure -> ensure

File: skipchain/struct.go [hunk 2]
```diff
@@ -767,6 +771,35 @@ func (db *SkipBlockDB) StoreBlocks(blocks []*SkipBlock) ([]SkipBlockID, error) {
...
+				// block is not yet discovered like in a catch up scenario that will end
+				// up in this path. But we can at least check that the forward-link belongs
```
Comment (kc1212) [inline skipchain/struct.go]: I understand the first sentence until "that will end up in this path". Can you rephrase it? Or remove the final part it if it's not super important.

Comment (Gilthoniel) [reply skipchain/struct.go]: I removed it, I guess it's more confusing than informative.
